prov/sockets/src/sock_progress.c

```diff
@@ -2323,6 +2327,12 @@ static int sock_pe_new_tx_entry(struct sock_pe *pe, struct sock_tx_ctx *tx_ctx)
 					msg_hdr->msg_len += datatype_sz *
 						pe_entry->pe.tx.tx_iov[i].src.ioc.count;
 			}
+			for (i = 0; i < pe_entry->pe.tx.tx_op.atomic.cmp_iov_len; i++) {
+				rbread(&tx_ctx->rb, &pe_entry->pe.tx.tx_iov[i].cmp,
+				 	sizeof(pe_entry->pe.tx.tx_iov[i].cmp));
+				msg_hdr->msg_len += datatype_sz *
+					pe_entry->pe.tx.tx_iov[i].cmp.ioc.count;
+			}
```

Comment:
Note that while sending out the atomic op request, the compare data is read from the iovecs. This should be updated to read from the inject buffer (+ offset).